### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by ankitt26**
#### The Space Travelers' Hub consists of Rockets, Missions, and the My Profile section.
- The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking.
- The Missions section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier.
- The My Profile section displays all reserved rockets and space missions.

#### Upon the first render fetch data from the SpaceX API endpoints:
 - Rockets: https://api.spacexdata.com/v4/rockets
 - Missions: https://api.spacexdata.com/v3/missions

### Checkpoints  ⛳
- [ ] Once the data are fetched, dispatch an action to store the selected data in Redux store.
- [ ] When a user clicks the "Reserve rocket" button dispatch action to update the store.
- [ ] When a user clicks the "Join Mission" button dispatch action to update the store.
- [ ] Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket.
- [ ] Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import Rockets from './components/Rockets';
+import Missions from './components/Missions';
+import { getMissions } from './Redux/missions/missionsSlice';
+import { fetchData } from './Redux/Rockets/rocketSlice';
+
+function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchData());
+    dispatch(getMissions());
```
Comment (AnselemOdims) [inline src/App.js:16]: - [ ] Good job on fetching the Rockets when the application starts, however, you are also fetching the missions. Please kindly ensure you only fetch missions once when you navigate to the missions page. You can sort this out by moving the dispatch action to the missions component.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [x] As required please kindly write tests using React Testing Library and Jest.
- [x] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/App.js (hunk 1 above)
Comment (DeliceLydia) [reply src/App.js]: Addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by ankitt26** [comment]
Thanks for approving 🎉.

**Comment by sunga12** [comment]
Thanks for Approving! 🥳🥳

**Pull request merged**